Add Lua tests to CI

vadi2 opened the pull request:
<!-- Keep the title short & concise so your mom can understand it -->
#### Brief overview of PR changes/additions
This PR completes the work of running our nearly 200 [Busted](http://olivinelabs.com/busted/)-based tests to run in CI automatically on every change, and complements the existing C++ tests which are run automatically already.
#### Motivation for adding to Mudlet
Better Mudlet quality!
#### Other info (issues closed, discussion etc)
We need to add a lot more tests to Lua to cover many aspects of the API =)

vadi2 committed:
Add a Ubuntu build

vadi2 committed:
Install homebrew dependencies only on macOS

vadi2 committed:
Add Linux dependencies separate

vadi2 committed:
Don't stop other builds on fail

vadi2 committed:
Install libzip-dev on Linux

vadi2 committed:
Sudo please

vadi2 committed:
Try installing libglu1-mesa-dev

vadi2 committed:
Install Pulse dev libraries

vadi2 committed:
Add a workaround for Lua linking

vadi2 committed:
Correct YAML

vadi2 committed:
Back to a single line

vadi2 committed:
Debug locations of libraries

vadi2 committed:
Correct an extra /

vadi2 committed:
Build Mudlet with tracing

vadi2 committed:
Try -GNinja Multi-Config

vadi2 committed:
Try not specifying a generator

keneanung committed:
Add link information against libdl.so as well (#33)

* Add link information against libdl.so as well

* Remove (harmful?) lua library path definition

vadi2 committed:
Show built output in the end

vadi2 committed:
Added a run tests flag

vadi2 committed:
Try appending gcc argument as well

vadi2 committed:
Debug plugin loading

vadi2 committed:
Use a single, generic compiler line

vadi2 committed:
Remove the x's

vadi2 committed:
Install libxkbcommon-x11-0

vadi2 committed:
Try launching Achaea profile

vadi2 committed:
Merge commit '313dcbb8cd4d7152fa2125c04ecb37fd2739f0ff' into add-busted-in-ci

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

# Conflicts:
#	.github/workflows/build-mudlet.yml

demonnic commented on .github/workflows/build-mudlet.yml:
```diff
@@ -85,3 +85,9 @@ jobs:
+      if: runner.os == 'Linux'
+      uses: GabrielBB/xvfb-action@v1.0
+      with:
+        run: ${{runner.workspace}}/b/ninja/src/mudlet -profile Achaea
```
```suggestion
        run: ${{runner.workspace}}/b/ninja/src/mudlet --profile Achaea
```

demonnic commented:
I have some changes to the run test xml I need to get pushed up to git, will try to do that today

vadi2 committed:
Update .github/workflows/build-mudlet.yml

Co-authored-by: Damian Monogue <3660+demonnic@users.noreply.github.com>

vadi2 committed:
Try taking a screenshot of what's running

vadi2 committed:
Fix YAML syntax

vadi2 committed:
Clarify the step name

vadi2 committed:
Take a screenshot even on build 'time out'

vadi2 committed:
Install imagemagick

vadi2 commented:
Will re-open when ready.

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

# Conflicts:
#	.github/workflows/build-mudlet.yml

vadi2 committed:
Fix whitespaces

vadi2 committed:
Upgrade to latest 1.5 of the xvfb action

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

vadi2 committed:
Pre-set environment variables to be used

vadi2 committed:
Increase the timeout

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

vadi2 committed:
Run with the self-test profile

vadi2 committed:
Re-enable mudlet.org's connection info

vadi2 committed:
Mock test runs

vadi2 committed:
Add an installing debug echo

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

vadi2 committed:
Debug with tmate

vadi2 committed:
Show error when script couldn't be loaded

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

vadi2 committed:
Print all script compile errors

vadi2 committed:
Setup tmate at the end

vadi2 commented:
We're getting a lot closer. @demonnic will take a look at running the tests inside Mudlet and I'm close to having Mudlet run and close on its own.

One blocker with the latter is that no Lua functions seem to be available in this environment - which I'm investigating.

vadi2 committed:
Add debug echoes

vadi2 committed:
Move Mudlet to the right location

vadi2 commented:
OK, got that squared away as well, everything is ready for the tests.

vadi2 committed:
Merge branch 'add-busted-in-ci' of https://github.com/Mudlet/Mudlet into add-busted-in-ci

vadi2 committed:
Split out packaging steps

vadi2 committed:
Cleanup

vadi2 committed:
Auto-run tests

vadi2 committed:
Install busted as well

vadi2 committed:
Fix setup() call

vadi2 committed:
Log everything sent to all consoles

vadi2 committed:
Actually log everything printed

vadi2 committed:
Pass in the test location

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

vadi2 committed:
Comment out a crash

vadi2 committed:
Tidier console print message

vadi2 committed:
Echo cleanup

vadi2 committed:
Better echo style

vadi2 committed:
Wrap closeMudlet in a timer to avoid a crash

vadi2 committed:
Comment out mSpellDic crash as well

vadi2 committed:
Set environment variable when test failed

vadi2 committed:
Fail workflow when tests fail

vadi2 committed:
Write to GITHUB_ENV properly

vadi2 committed:
Add setEnv

vadi2 committed:
Make use of setEnv

vadi2 committed:
Launch tmate

vadi2 committed:
Write to file instead

vadi2 committed:
Add comments

vadi2 committed:
Improve error message

vadi2 committed:
Remove tmate

vadi2 commented:
I've got the final part squared as away as well. [It works](https://github.com/Mudlet/Mudlet/runs/3451872086?check_suite_focus=true#step:27:34)!


![image](https://user-images.githubusercontent.com/110988/131244062-4f0b75ae-62ec-49e2-b78c-8349a557475e.png)

Expanding the previous step, you see the issue...

![image](https://user-images.githubusercontent.com/110988/131244065-e6ca837a-f5ac-4f7f-afdf-a92eab17aea2.png)

😃

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

vadi2 committed:
Fix merge error

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-busted-in-ci

vadi2 committed:
Add the new --mirror option

vadi2 committed:
Don't deploy Windows builds

vadi2 committed:
Only check Lua tests when we need to

vadi2 committed:
Update job name

vadi2 committed:
Remove unusued setEnv

vadi2 committed:
Merge branch 'add-busted-in-ci' of https://github.com/Mudlet/Mudlet into add-busted-in-ci

demonnic reviewed (approved):
yay!

vadi2 commented:
@Chris7

demonnic commented:
hmm, it seems to think there's a status still missing. If ubuntu/gcc replaced by or supplemented by ubuntu/gcc/lua tests?

vadi2 commented:
Replaced - github was confused because we marked `ubuntu / gcc` as a mandatory one in settings, but it won't run in this workflow anymore. I've removed it for the time being, and after we merge this PR, we can add `ubuntu / gcc / lua tests` as a mandatory one back again.

It'll mean that any existing PRs will need to be updated with `development` once that happens, but that is simple to do.

demonnic commented:
ok cool, that's what I thought might be the case.

vadi2 commented:
@LiamLeFey
